x-pack/plugin/rank-rrf/src/test/java/org/elasticsearch/xpack/rank/rrf/RRFRankDocTests.java

```diff
@@ -46,34 +51,49 @@ protected RRFRankDoc createTestInstance() {
 
     @Override
     protected RRFRankDoc mutateInstance(RRFRankDoc instance) throws IOException {
-        RRFRankDoc mutated = new RRFRankDoc(instance.doc, instance.shardIndex, instance.positions.length);
-        mutated.score = instance.score;
-        mutated.rank = instance.rank;
-        System.arraycopy(instance.positions, 0, mutated.positions, 0, instance.positions.length);
-        System.arraycopy(instance.scores, 0, mutated.scores, 0, instance.positions.length);
-        mutated.rank = mutated.rank == NO_RANK ? randomIntBetween(1, 10000) : NO_RANK;
-        if (rarely()) {
-            int ri = randomInt(mutated.positions.length - 1);
-            mutated.positions[ri] = mutated.positions[ri] == NO_RANK ? randomIntBetween(1, 10000) : NO_RANK;
-        }
-        if (rarely()) {
-            int ri = randomInt(mutated.positions.length - 1);
-            mutated.scores[ri] = randomFloat();
-        }
-        if (rarely()) {
-            mutated.doc = randomNonNegativeInt();
-        }
-        if (rarely()) {
-            mutated.score = randomFloat();
-        }
-        if (frequently()) {
-            mutated.shardIndex = mutated.shardIndex == -1 ? randomNonNegativeInt() : -1;
+        int doc = instance.doc;
+        int shardIndex = instance.shardIndex;
+        float score = instance.score;
+        int rankConstant = instance.rankConstant;
+        int rank = instance.rank;
+        int queries = instance.positions.length;
+        int[] positions = new int[queries];
+        float[] scores = new float[queries];
+
+        switch (randomInt(6)) {
+            case 0:
+                doc = randomNonNegativeInt();
```

Comment:
To be doubly sure, you likely need to do `randomValueOtherThan(doc, () -> randomNonNegativeInt());` for all of these. Though the likelihood is really low :)

same goes for all the other values pretty much. But, it might not be necessary.

Reply:
Updated :)